scripts/mbedtls_framework/c_wrapper_generator.py

```diff
@@ -30,16 +30,28 @@ def c_declare(prefix: str, name: str, suffix: str) -> str:
     ('wrapper_name', str),
 ])
 
-def strip_indentation(in_str: str, new_lines: int = 1, ident_lv: int = 4) -> str:
-    """Return a whitespace stripped str, appended with user defined empty lines.
+def strip_indentation(in_str: str, new_lines: int = 1, ident_lv: int = 0) -> str:
+    """Return a whitespace stripped str, with configurable whitespace in output.
 
-    The method will remove (ident_lv * 4) space-character identations from
-    input string. It will also remove all whitespace around the text-block.
+    The method will remove space-character identations from input string.
```

Comment:
Given the following input:
```
  foo
   bar
 qux
```

The documentation says that indentation is stripped on each line, so the output is:
```
foo
bar
qux
```

The actual output strips the indentation of the first line with a non-zero indent (which is pretty bizarre):
```
foo
 bar
 qux
```

The intent is to strip the common indentation, so the output should be:
```
 foo
  bar
qux
```

Please fix. You can call `textwrap.dedent`.

Reply:
Since the method is already reading the identation levels `    _input_idents = [len(n) for n in re.findall(r'(?m)^ +', in_str)]` a call to `min(_input_idents )` will be sufficent to achieve that formatting. 

Will update